include/rwqueue.h

```diff
@@ -51,7 +51,11 @@ class RWQueue {
 	size_t MaxCapacity() const;
 	int GetPercentFull();
 
-	void Enqueue(const T& item);
+	// Discourage copying into the queue. Instead, use std::move into the
+	// queue to explicitly invalidate the source object to avoid having
+	// two source objects floating around.
+	//
+	// void Enqueue(const T& item);
```

Comment:
Nice! 🎉  Hopefully this approach will result in less performance-related surprises!